ts/utils/navigation.ts

```diff
@@ -2,54 +2,46 @@
 
 import { index } from "fp-ts/lib/Array";
 import { fromNullable, none, Option } from "fp-ts/lib/Option";
-import { NavigationRoute } from "react-navigation";
+import {
+  NavigationLeafRoute,
+  NavigationRoute,
+  NavigationState
+} from "react-navigation";
 import { NavigationHistoryState } from "../store/reducers/navigationHistory";
 
-// TODO: Need to be fixed https://www.pivotaltracker.com/story/show/170819360
-export function getCurrentRouteName(navNode: any): string | undefined {
-  if (!navNode) {
-    return undefined;
-  }
-  if (
-    navNode.index === undefined &&
-    navNode.routeName &&
-    typeof navNode.routeName === "string"
-  ) {
-    // navNode is a NavigationLeafRoute
-    return navNode.routeName;
+/**
+ * Assuming that each and every NavigationRoute will eventually lead
+ * to a NavigationLeafRoute.
+ */
+const findLeafRoute = (
+  branchOrLeaf: NavigationRoute
+): NavigationLeafRoute | undefined => {
+  const { routes, index } = branchOrLeaf;
+  if (routes) {
```

Comment:
```suggestion
  if (routes.length > index) {
```
could `routes` be `undefined`?
if not an empty array (`[]`) passes the check and `findLeafRoute` could be invoked with an `undefined` parameter

Reply:
This is a tough topic 😢 

**Theoretically** the navigation engine guarantees that exists an `index` and a `routes` array, and that `routes[index]` is defined. This consistency is however handled by them and is not a type constraint.
This means that if `routes` exists, must not be empty.

This is why the type of `findeLeafRoute` looks inconsistent as [you correctly pointed out](https://github.com/pagopa/io-app/pull/3329#discussion_r699458595).

On top of that, our tests break these functions by creating (apparently) an inconsistent state, and that's why [the try/catch](https://github.com/pagopa/io-app/pull/3329/files#diff-62de5eb9c9c613534738074b70728d9fd0631c7f98455ef9e774e189252b23bbR30) is needed. I haven't gone in depth because we are going to upgrade React Navigation anyway and it will be a major change.

What I believe makes sense to do now is to improve the implementation to make explicit the inconsistency: see https://github.com/pagopa/io-app/pull/3329/commits/464176f6797146f79209e4beebe916b43a1346b5